[utils] Print compilation time breakdown across main components

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #151410
* #151409
* #150704
* __->__ #150717
* #151357
* #151256
* #151330

Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

anijain2305 committed:
[utils] Print compilation time breakdown across main components

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="320" alt="image" src="https://github.com/user-attachments/assets/3095a1a5-a260-44b6-b8d2-e87bfeaa2779" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -776,14 +776,65 @@ def fmt_fn(values, item_fn=lambda x: x):
+        # Calculate approximate component breakdown
+        # Triton time
+        # These triton times are for both fwd and bwd
+        triton_kernels = get_metric("async_compile.wait")
```
Man, I wish this was better, but we also seem to have "async_compile.precompile" in the case that parallel compile is not active and we do the compilation in the main process :/

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -776,14 +776,65 @@ def fmt_fn(values, item_fn=lambda x: x):
+        inductor_total = inductor_fwd_triton + inductor_bwd_triton - triton_kernels
+
+        # Dynamo time
+        entire_frame_fwd = get_metric("_compile.compile_inner")
```
We also have "backward._backward_impl" in runtime_wrappers.py, which I think captures the total compile time for the backward??

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -776,14 +776,65 @@ def fmt_fn(values, item_fn=lambda x: x):
+            entire_frame_fwd + inductor_bwd_triton + triton_autotuning - accounted
+        )
+
+        if accounted:
```
accounted or unaccounted?

masnesral reviewed (approved):
There are a couple more dynamo_timed callsites I think you might want

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

anijain2305 committed:
Update on "[utils] Print compilation time breakdown across main components"


Prints something like this 
<img width="384" alt="image" src="https://github.com/user-attachments/assets/eaeae3ec-bab1-42e2-acbf-9e74904c6ac2" />

which is very helpful for quick compile time optimizations testing.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]
